internal/partinfo/ext.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2023 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package partinfo
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+)
+
+const (
+	extMagic uint16 = 0xEF53 // Ext magic signature
+)
+
+type extPartition struct {
+	f          *os.File
+	superblock struct {
+		_       [56]byte
+		Magic   uint16 // Ext magic signature
+		_       [62]byte
+		VolName [16]byte // Volume name
+	}
+}
+
+func (p extPartition) Path() string {
+	return p.f.Name()
+}
+
+func (p extPartition) FSType() string {
+	// TODO: Could this lead to issues when attempting to mount ext2/3 partitions?
+	return "ext4"
+}
+
+func (p extPartition) Label() string {
+	return strings.TrimRight(string(p.superblock.VolName[:]), "\x00")
+}
+
+func newExtPartition(f *os.File) (Partition, error) {
```

Comment:
I realize these are internal functions, but it'd be slightly better (for testability etc) if these functions took an [`io.ReadSeeker`](https://pkg.go.dev/io#ReadSeeker) instead of a concrete `*os.File` (which implements `io.ReadSeeker`).

Similarly, it should probably return a concrete `extPartition` (which implements `Partition`). The pattern in Go is very often "accept interfaces, return structs".

Reply:
https://github.com/anpep/pebble/blob/05f9184528ed8c905aff6a00c61258fdc0101a94/internal/partinfo/ext.go#L40-L42

Reply:
I guess you could pass the path in separately. But yeah, this is fine -- concrete is simpler if we can get away with it. Files are easy to create in tests anyway. :-)